File: clients/fbaModelServicesClient.py
try:
    import json
except ImportError:
    import sys
    sys.path.append('simplejson-2.3.3')
    import simplejson as json
    
import urllib



class fbaModelServices:

    def __init__(self, url):
        if url != None:
            self.url = url

    def genome_to_fbamodel(self, in_genome):

        arg_hash = { 'method': 'fbaModelServices.genome_to_fbamodel',
                     'params': [in_genome],
                     'version': '1.1'
                     }

        body = json.dumps(arg_hash)
        resp_str = urllib.urlopen(self.url, body).read()
        resp = json.loads(resp_str)

        if 'result' in resp:
            return resp['result'][0]
        else:
            return None

    def fbamodel_to_exchangeFormat(self, in_model):

        arg_hash = { 'method': 'fbaModelServices.fbamodel_to_exchangeFormat',
                     'params': [in_model],
                     'version': '1.1'
                     }

        body = json.dumps(arg_hash)
        resp_str = urllib.urlopen(self.url, body).read()
        resp = json.loads(resp_str)

        if 'result' in resp:
            return resp['result'][0]
        else:
            return None

    def exchangeFormat_to_fbamodel(self, in_model):

        arg_hash = { 'method': 'fbaModelServices.exchangeFormat_to_fbamodel',
                     'params': [in_model],
                     'version': '1.1'
                     }

        body = json.dumps(arg_hash)
        resp_str = urllib.urlopen(self.url, body).read()
        resp = json.loads(resp_str)

        if 'result' in resp:
            return resp['result'][0]
        else:
            return None

    def fbamodel_to_sbml(self, in_model):

        arg_hash = { 'method': 'fbaModelServices.fbamodel_to_sbml',
                     'params': [in_model],
                     'version': '1.1'
                     }

        body = json.dumps(arg_hash)
        resp_str = urllib.urlopen(self.url, body).read()
        resp = json.loads(resp_str)

        if 'result' in resp:
            return resp['result'][0]
        else:
            return None

    def sbml_to_fbamodel(self, in_model):

        arg_hash = { 'method': 'fbaModelServices.sbml_to_fbamodel',
                     'params': [in_model],
                     'version': '1.1'
                     }

        body = json.dumps(arg_hash)
        resp_str = urllib.urlopen(self.url, body).read()
        resp = json.loads(resp_str)

        if 'result' in resp:
            return resp['result'][0]
        else:
            return None

    def gapfill_fbamodel(self, in_model, in_formulation, overwrite, save):

        arg_hash = { 'method': 'fbaModelServices.gapfill_fbamodel',
                     'params': [in_model, in_formulation, overwrite, save],
                     'version': '1.1'
                     }

        body = json.dumps(arg_hash)
        resp_str = urllib.urlopen(self.url, body).read()
        resp = json.loads(resp_str)

        if 'result' in resp:
            return resp['result'][0]
        else:
            return None

    def runfba(self, in_model, in_formulation, overwrite, save):

        arg_hash = { 'method': 'fbaModelServices.runfba',
                     'params': [in_model, in_formulation, overwrite, save],
                     'version': '1.1'
                     }

        body = json.dumps(arg_hash)
        resp_str = urllib.urlopen(self.url, body).read()
        resp = json.loads(resp_str)

        if 'result' in resp:
            return resp['result'][0]
        else:
            return None

    def object_to_html(self, inObject):

        arg_hash = { 'method': 'fbaModelServices.object_to_html',
                     'params': [inObject],
                     'version': '1.1'
                     }

        body = json.dumps(arg_hash)
        resp_str = urllib.urlopen(self.url, body).read()
        resp = json.loads(resp_str)

        if 'result' in resp:
            return resp['result'][0]
        else:
            return None




        

File: clients/fbaModelServicesClient.cs



namespace KBase.CDMI
{
    class Client
    {
        private JsonRpcClient rpc_client;
        private string url;

        public CDMI_Client(string url )
        {
            this.url = url;
            rpc_client = new JsonRpcClient();
            rpc_client.Url = url;
            rpc_client.RequestEncoding = new UTF8Encoding(false);
        }
       

        public Object genome_to_fbamodel(object in_genome)
        {
            object[] plist = new object[] { in_genome } ;

            Object ret = rpc_client.Invoke("fbaModelServices.genome_to_fbamodel", plist);
            return ret;
        }

        public Object fbamodel_to_sbml(object in_model)
        {
            object[] plist = new object[] { in_model } ;

            Object ret = rpc_client.Invoke("fbaModelServices.fbamodel_to_sbml", plist);
            return ret;
        }

        public Object gapfill_fbamodel(object in_model, object in_formulation, object overwrite, object save)
        {
            object[] plist = new object[] { in_model, in_formulation, overwrite, save } ;

            Object ret = rpc_client.Invoke("fbaModelServices.gapfill_fbamodel", plist);
            return ret;
        }

        public Object runfba(object in_model, object in_formulation, object overwrite, object save)
        {
            object[] plist = new object[] { in_model, in_formulation, overwrite, save } ;

            Object ret = rpc_client.Invoke("fbaModelServices.runfba", plist);
            return ret;
        }

        public Object object_to_html(object inObject)
        {
            object[] plist = new object[] { inObject } ;

            Object ret = rpc_client.Invoke("fbaModelServices.object_to_html", plist);
            return ret;
        }

        public Object gapgen_fbamodel(object in_model, object in_formulation, object overwrite, object save)
        {
            object[] plist = new object[] { in_model, in_formulation, overwrite, save } ;

            Object ret = rpc_client.Invoke("fbaModelServices.gapgen_fbamodel", plist);
            return ret;
        }


    }
